src/main/java/org/dataone/hashstore/hashfs/HashFileStore.java

```diff
@@ -2,26 +2,47 @@
 
 import java.io.File;
 import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.AtomicMoveNotSupportedException;
 import java.nio.file.FileAlreadyExistsException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
 /**
- * HashFileStore handles IO operations for HashStore
+ * HashFileStore is a class that enables the storing of objects to disk by
+ * utilizing the SHA-256 hex digest of an authority-based identifier as a key,
+ * which typically takes the form of a persistent identifier (pid). This class
+ * is responsible for managing the underlying storage implementation, as well as
+ * providing an interface for interacting with stored objects and their
+ * associated metadata.
+ *
  */
 public class HashFileStore {
+    private static final Log logHashFileStore = LogFactory.getLog(HashFileStore.class);
     private final int directoryDepth;
     private final int directoryWidth;
     private final String objectStoreAlgorithm;
     private final Path objectStoreDirectory;
     private final Path tmpFileDirectory;
-    private HashUtil hashUtil = new HashUtil();
+    public String[] supportedHashAlgorithms = { "MD2", "MD5", "SHA-1", "SHA-256", "SHA-384", "SHA-512", "SHA-512/224",
```

Comment:
I think the variable is a constance. If this is the case, we usually declare it static and final. Also we use capital letters as the name.

Reply:
Thank you! I've updated the variable

Reply:
> Are these supported algos determined by the backing store in any way? Or should they really be part of `HashStore`?

These algorithms are supported by the `java.security.MessageDigest` class - which can technically be implemented by any class. Depending on the outcome from our discussion regarding `HashStore` vs `HashFileStore` implementing an interface, this may or may not shift.

Reply:
@doulikecookiedough Let's not hardcode the list of algorithms, and instead make that a configuration item that can be changed without having to recompile code.